### PR TITLE
Kamon integration

### PR DESCRIPTION
**Pull request by helgridly**
This isn't really DSDEEPB-2034 but it's the closest issue we have
- [x] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [x] Tell  that the PR exists if he wants to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
- [ ] **LR**: Initial review by LR and others.
- [ ] Comment / review / update cycle:
  - Rest of team may comments on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter updates documentation as needed
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH. **Reassign to LR** for further feedback
- [x]  sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Squash commits, rebase if necessary
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Merge to develop 
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: Check configuration files in Jenkins in case they need changes
- [ ] **Submitter**: Verify swagger UI on dev server still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by helgridly** [comment]
hey @dvoet i integrated kamon

File: build.sbt [hunk 1]
```diff
@@ -45,22 +53,47 @@ libraryDependencies ++= {
       excludeAll (ExclusionRule(organization = "com.gettyimages"), ExclusionRule(organization = "org.webjars")) ,
     "org.broadinstitute.dsde.vault" %% "vault-common" % "0.1-15-bf74315",
     ("com.google.apis" % "google-api-services-storage" % "v1-rev30-1.20.0").exclude("com.google.guava", "guava-jdk5"),
-    ("com.google.apis" % "google-api-services-compute" % "v1-rev72-1.20.0"),
-    ("com.google.apis" % "google-api-services-admin-directory" % "directory_v1-rev53-1.20.0")
+    "com.google.apis" % "google-api-services-compute" % "v1-rev72-1.20.0",
```
Comment (jmthibault79) [inline build.sbt:56]: Does this have any meaningful effect, or is it just for consistency?

File: src/main/resources/application.conf [hunk 2]
```diff
@@ -31,3 +31,33 @@ submission-monitor-dispatcher {
...
+    }
+
+    akka-router {
+      includes = [ "**" ]
+      excludes = []
+    }
+  }
+
+  statsd {
+    // If you're using Kamon instrumentation you'll want need to specify your hostname (and potentially port) where
```
Comment (jmthibault79) [inline src/main/resources/application.conf:57]: want or need?

File: build.sbt (hunk 1 above)
Comment (helgridly) [inline build.sbt:56]: Consistency.

File: src/main/resources/application.conf (hunk 2 above)
Comment (helgridly) [inline src/main/resources/application.conf:57]: I copied this from Cromwell :(

Comment (geoffjentry) [inline src/main/resources/application.conf:57]: Which in turn was largely copied from Agora.

If you leave it blank it defaults to 127.0.0.1

If you give it some bogus value it won't crash but it won't do anything useful either

Comment (helgridly) [inline src/main/resources/application.conf:57]: Like Cromwell, I'm planning to add the value to Jenkins.

**Comment by geoffjentry** [comment]
If you have actors that you want to group by class (e.g. PerRequest type stuff where there are a crapton of individual instances) you'll need to make some rewrite rules. Easier to explain in person, but it's not a code change for rawls

**Comment by jmthibault79** [comment]
Do you mean that we can get away without wrapping a `traceName()` around each route?  I'd prefer that, if possible.

File: src/main/resources/application.conf (hunk 2 above)
Comment (jmthibault79) [inline src/main/resources/application.conf:57]: Vault / Docker template

**Comment by geoffjentry** [comment]
@jmthibault79 No. What I mean is that for instance in Cromwell we have a class WorkflowActor. But there's one of those for every single workflow ever run, and each one will normally appear as a unique entity in statsd and thus on the dashboard. That'd become useless quickly. So what I did was make a rewrite rule to convert all WorkflowActor-blahblahblah to just WorkflowActor, effectively aggregating them.

If you're naming your actors (and you should) a similar effect will happen for each endpoint's PerRequest actor

**Comment by geoffjentry** [comment]
Rewrite rule examples are here: https://github.com/broadinstitute/dsde-dashboard/blob/develop/src/main/resources/graphite/rewrite-rules.conf

The not at all documented trick here is that _all_ of the rules are applied in order so make sure to not have fairly generic rules as the results might be confusing

File: src/main/resources/application.conf (hunk 2 above)
Comment (helgridly) [inline src/main/resources/application.conf:57]: Oh right, yeah. Thanks for the catch!

**Comment by jmthibault79** [comment]
No additional review comments from me.  Hooray for instrumentation!

**Comment by dvoet** [comment]
:+1:

**Pull request closed without merging**